src/main/java/sim/model/Server.java

```diff
@@ -77,16 +72,22 @@ public long getSpeed() {
     // Subject methods
     // add observers to a list
 	public void registerObserver(Observer o) {
-		observers.add(o);
+		LinkedList<Observer> current = new LinkedList<>();
+		current = (LinkedList)observers.clone(); // avoids concurrent modification exception if notifyObserver() is called while observers are being added
+		current.add(o);
+		observers = current;
 	}
 
 	// removes observers from a list
 	public void removeObserver(Observer o) {
-		observers.remove(o);
+		LinkedList<Observer> current = new LinkedList<>();
+		current = (LinkedList)observers.clone(); // avoids concurrent modification exception if notifyObserver() is called while observers are being added
```

Comment:
I'd maybe move the comments to the line above just for readability (avoid long line length)